validation-test/Sema/type_checker_crashers_fixed/issue59294.swift

```diff
@@ -25,18 +26,94 @@ struct WrapperValue<Value> {
   func printValue() {
     print(value)
   }
+
+  func returnValue() -> Value {
+    return value
+  }
 }
 
+@propertyWrapper
+struct OuterWrapper<Value> {
+  var value: Value
+  init(wrappedValue: Value) {
+    self.value = wrappedValue
+  }
+
+  var projectedValue: Self {
+    return self
+  }
+
+  var wrappedValue: Value {
+    get {
+      self.value
+    }
+    set {
+      self.value = newValue
+    }
+  }
+}
 
 class Test {
   static func test() {
     return [0, 1, 2].compactMap { _ in // expected-error {{unexpected non-void return value in void function}} expected-note {{did you mean to add a return type?}}
       @WrapperValue var value: Bool? = false
       if value != nil {
+        $value.printValue()
         return false
       }
 
       return value ?? false
     }
   }
+  static func testProjectedAndWrapperVars() {
+    func test(_: (Int) -> Bool) {}
+    func test(_: (String) -> Void) {}
+
+    test {
+      @WrapperValue var value = $0
+      if $value.returnValue() > 0 {
+        test {
+          return $value.returnValue() == $0 &&
+                 _value == $0
+        }
+      }
+      return false
+    }
+
+    test {
+      @WrapperValue var value = $0
+
+      test {
+        if _value != $0 { // Ok
+          $value.printValue()
+        }
+      }
+    }
+  }
+
+  static func testNestedWrappers() {
+    func test(_: (Bool) -> Void) {}
+    func test(_: () -> Void) {}
+
+    test {
+      if true {
+        @OuterWrapper @WrapperValue var value = $0
+        if true {
+          let _: Bool = _value == $0
+          let _: OuterWrapper<Bool> = $value
```

Comment:
The type of the projected value here should be `OuterWrapper<WrapperValue<Bool>>`

Reply:
Fixed by using correct "outer" wrapper, thank you!